src/event.rs

```diff
@@ -222,51 +222,86 @@ pub enum WindowEvent {
-    /// [`transform`]: https://developer.mozilla.org/en-US/docs/Web/CSS/transform
-    CursorEntered { device_id: DeviceId },
+    PointerEntered {
+        device_id: DeviceId,
+
+        /// The position of the cursor when it entered the window.
+        ///
+        /// ## Platform-specific
+        ///
+        /// **Orbital: Always emits `(0., 0.)`.
+        /// **Web:** Doesn't take into account CSS [`border`], [`padding`], or [`transform`].
+        ///
+        /// [`border`]: https://developer.mozilla.org/en-US/docs/Web/CSS/border
+        /// [`padding`]: https://developer.mozilla.org/en-US/docs/Web/CSS/padding
+        /// [`transform`]: https://developer.mozilla.org/en-US/docs/Web/CSS/transform
+        position: PhysicalPosition<f64>,
+
+        ty: PointerType,
+    },
 
     /// The cursor has left the window.
-    ///
-    /// ## Platform-specific
-    ///
-    /// - **Web:** Doesn't take into account CSS [`border`], [`padding`], or [`transform`].
-    ///
-    /// [`border`]: https://developer.mozilla.org/en-US/docs/Web/CSS/border
-    /// [`padding`]: https://developer.mozilla.org/en-US/docs/Web/CSS/padding
-    /// [`transform`]: https://developer.mozilla.org/en-US/docs/Web/CSS/transform
-    CursorLeft { device_id: DeviceId },
+    PointerLeft {
+        device_id: DeviceId,
+
+        /// The position of the cursor when it left the window.
+        ///
+        /// ## Platform-specific
+        ///
+        /// **Orbital/Windows:** Always emits [`None`].
+        /// **Web:** Doesn't take into account CSS [`border`], [`padding`], or [`transform`].
+        ///
+        /// [`border`]: https://developer.mozilla.org/en-US/docs/Web/CSS/border
+        /// [`padding`]: https://developer.mozilla.org/en-US/docs/Web/CSS/padding
+        /// [`transform`]: https://developer.mozilla.org/en-US/docs/Web/CSS/transform
+        position: Option<PhysicalPosition<f64>>,
+
+        ty: PointerType,
+    },
 
     /// A mouse wheel movement or touchpad scroll occurred.
     MouseWheel { device_id: DeviceId, delta: MouseScrollDelta, phase: TouchPhase },
 
     /// An mouse button press has been received.
-    MouseInput { device_id: DeviceId, state: ElementState, button: MouseButton },
+    PointerButton {
+        device_id: DeviceId,
+        state: ElementState,
+
+        /// The position of the pointer when the button was pressed.
+        ///
+        /// ## Platform-specific
+        ///
+        /// - **Orbital: Always emits `(0., 0.)`.
+        /// - **Web:** Doesn't take into account CSS [`border`], [`padding`], or [`transform`].
+        ///
+        /// [`border`]: https://developer.mozilla.org/en-US/docs/Web/CSS/border
+        /// [`padding`]: https://developer.mozilla.org/en-US/docs/Web/CSS/padding
+        /// [`transform`]: https://developer.mozilla.org/en-US/docs/Web/CSS/transform
+        position: PhysicalPosition<f64>,
+
+        button: ButtonSource,
```

Comment:
I _feel_ like I'd rather mirror web even closer here, and have:
```suggestion
        button: Option<MouseButton>,
        source: PointerSource,
```

Instead of creating a whole `ButtonSource` which ends up containing much of the same information. But that may be an oversimplication, and perhaps it doesn't align well with future plans for pointer events?

Reply:
Yes, its needed for future pointer kinds.
Its also important that users can handle any pointer type, this is why `ButtonSource::mouse_button()` exists.
If we change it to `Option`, users will think no button was pressed on other pointer types, which would be incorrect. E.g. touch does indeed mimic the `MouseButton::Left` and so does regular pen input.

> I _feel_ like I'd rather mirror web even closer here, [..]

Actually the current API is how the Web API does this.
For touch and pen the same ID as the left mouse button is used, see the [W3C specification](https://w3c.github.io/pointerevents/#the-button-property).

Reply:
I'd feel even better with `button: MouseButton`!

But it's also fine to me if it makes sense for future pointer kinds (pens?), I'll trust your judgement on this (depending on how long you think it'll take before we get it implemented, if it'll take long, then I'd prefer the solution with fewer newtypes to start with).

Reply:
I agree!

My plan is to do it right after, if for some reason or another this doesn't happen, I promise to make a PR to address this!